mm2src/coins/eth/v2_activation.rs

```diff
@@ -660,9 +664,12 @@ pub(crate) async fn build_address_and_priv_key_policy(
                 .mm_err(|e| EthActivationV2Error::InternalError(e.to_string()))?;
             let activated_key = KeyPair::from_secret_slice(raw_priv_key.as_slice())
                 .map_to_mm(|e| EthActivationV2Error::InternalError(e.to_string()))?;
+            #[cfg(not(target_arch = "wasm32"))]
+            run_db_migration_for_new_eth_pubkey(ctx, &activated_key).await?;
+
             let bip39_secp_priv_key = global_hd_ctx.root_priv_key().clone();
 
-            let hd_wallet_rmd160 = *ctx.rmd160();
+            let hd_wallet_rmd160 = dhash160(activated_key.public().as_bytes());
```

Comment:
tbh I am still lost in here. The previous method of deciding `hw_wallet_rmd160` made sense (it depended only on the used HD account, thus it's shared, so using different address within the same account will yield the same rmd160).

right now it looks like we get different shared db for each address in the HD account. why is it shared then?
we are using the activated pubkey directly which means if you use different addresses within the same account they won't share the shared DB.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2093/commits/0b4d808e59901cecc7ed4ca46a817c1cada24efd